### PR TITLE
CI builds for additional C++ standards

### PR DESCRIPTION
**Pull request by offa**
CI Builds for _gcc_ and _clang_ using _C++98/11/14/17_ (#1242). I intentionally skipped _C++03_ since it shouldn't differ from C++98 and the build list is already long. I can add it at any time if it's reasonable – no problem.

C++11 and above need to set the `C++11=ON` option to correct warning settings. If another C++ standard is passed via `CMAKE_CXX_STANDARD` it shouldn't set the compiler standard option though. If compilation with a newer standard than C++11 is something users typically do, we should rethink this.

### CODE REVIEW

**Comment by offa** [comment]
The DOS build failed with a strange ALSA issue.

File: .travis.yml [hunk 1]
```diff
@@ -5,22 +5,38 @@ dist: bionic
...
     - compiler: gcc
-      env: BUILD=cmake CPP11=OFF
+      env: BUILD=cmake CPP_STD=98 CPP11=OFF
+    - compiler: gcc
+      env: BUILD=cmake CPP_STD=11 CPP11=ON
+    - compiler: gcc
+      env: BUILD=cmake CPP_STD=14 CPP11=ON
     - compiler: gcc
-      env: BUILD=cmake CPP11=ON
+      env: BUILD=cmake CPP_STD=17 CPP11=ON
```
Comment (basvodde) [inline .travis.yml]: The CPP11=ON is a bit weird here :) Can we do something about that?

**Comment by basvodde** [comment]
Any idea about the DOS failure?

File: .travis.yml (hunk 1 above)
Comment (offa) [reply .travis.yml]: Indeed, this is horrible. I have updated the PR to take `CMAKE_CXX_STANDARD` into account when setting the warnings. If it's set with a standard >= C++11 it'll add the warnings accordingly, making the `CPP11` option no longer necessary.

**Comment by offa** [comment]
> Any idea about the DOS failure?

Seems like a temporary problem. Previous and current builds don't have this problem.

**Comment by basvodde** [comment]
Tnx!

**Pull request merged**